src/servers/Server.cc

```diff
@@ -204,8 +204,14 @@ Server::clientWriteDone(const CommIoCbParams &io)
 
     Must(io.conn->fd == clientConnection->fd);
 
-    if (io.flag && pipeline.front())
-        pipeline.front()->initiateClose("write failure");
+    if (io.flag) {
+        debugs(33, 2, "bailing after a write failure: " << xstrerr(io.xerrno));
+        LogTagsErrors lte;
+        lte.timedout = io.xerrno == ETIMEDOUT;
+        lte.aborted = !lte.timedout; // intentionally true for zero io.xerrno
```

Comment:
These three lines are copied from similar I/O error-handling code in Server::doClientRead(). I recommend that we keep this code duplication in this vulnerability-fixing PR and follow up with a cleanup PR that encapsulates this logic. That followup PR may also adjust doClientRead() to use ERR_READ_ERROR instead of the current less specific ERR_CLIENT_GONE.